File: Poo Youtube Kotlin/src/main/kotlin/Video.kt
class Video(Titulo: String): VideoActions{
    //Atributos:
    var titulo = Titulo
        protected set
    var avaliacao: String = "Neutra!"
        protected set
    var views: Int = 0
        protected set
    var curtidas: Int = 0
        protected set
    var dislikes: Int = 0
        protected set
    var quemcur: MutableList<String> = mutableListOf<String>()
        private set
    var quemncur: MutableList<String> = mutableListOf<String>()
        private set
    var reproduzindo: Boolean = false
        protected set

    //Metodos abstratos
    override fun play() {
        this.reproduzindo = true
        this.views += 1
    }

    override fun pause() {
        this.reproduzindo = false
    }

    override fun like(gafanhoto: Gafanhoto) {
        if (gafanhoto.login in this.quemcur) {
            println("Esse Gafanhoto já curtiu!")
        }else{
            this.quemcur.add(gafanhoto.login)
            this.curtidas += 1
        }
    }

    override fun dislike(gafanhoto: Gafanhoto) {
        if (gafanhoto.login in this.quemncur) {
            println("Esse Gafanhoto já deu dislike!")
        }else{
            this.quemncur.add(gafanhoto.login)
            this.dislikes += 1
        }
    }

    //Metodos publicos
    fun detalhesfilm(){
        if (curtidas > dislikes){
            this.avaliacao = "Positiva!"
        }else if (curtidas < dislikes){
            this.avaliacao = "Negativa!"
        }
        println("Titulo: $titulo, Views: $views, Avaliação: $avaliacao, $curtidas, $dislikes")
    }
}

File: POO Youtube Python/Video.py
from Gafanhoto import Gafanhoto


class Video:
    # Atributos
    __avaliacao = "Neutra"
    __views = 0
    __curtidas = 0
    __quemcur = []
    __quemncur = []
    __dislikes = 0
    __reproduzindo = False

    # Construtor
    def __init__(self, Titulo: str):
        self.__titulo = Titulo

    # Metodos publicos
    def play(self):
        self.__reproduzindo = True
        self.__views += 1

    def pause(self):
        self.__reproduzindo = False

    def like(self, Espectador: Gafanhoto):
        if Espectador.login in self.__quemcur:
            print("Gafanhoto já curtiu!")
        else:
            self.__quemcur.append(Espectador.login)
            self.__curtidas += 1

    def dislike(self, Espectador: Gafanhoto):
        if Espectador.login in self.__quemncur:
            print("Gafanhoto já deu dislike!")
        else:
            self.__quemncur.append(Espectador.login)
            self.__dislikes += 1

    # Metodos especiais
    def __repr__(self):
        if self.__curtidas > self.__dislikes:
            self.__avaliacao = "Positiva!"
        elif self.__curtidas < self.__dislikes:
            self.__avaliacao = "Negativa!"
        elif self.__curtidas == self.__dislikes:
            self.__avaliacao = "Neutra!"
        return f'{self.__titulo}, {self.__avaliacao}, {self.__views}'

    @property
    def avaliacao(self):
        return self.__avaliacao
